odoo/addons/base/models/res_users.py

```diff
@@ -1237,17 +1237,21 @@ def unlink(self):
 class UsersView(models.Model):
     _inherit = 'res.users'
 
-    @api.model
-    def create(self, values):
-        values = self._remove_reified_groups(values)
-        user = super(UsersView, self).create(values)
-        group_multi_company = self.env.ref('base.group_multi_company', False)
-        if group_multi_company and 'company_ids' in values:
-            if len(user.company_ids) <= 1 and user.id in group_multi_company.users.ids:
-                user.write({'groups_id': [(3, group_multi_company.id)]})
-            elif len(user.company_ids) > 1 and user.id not in group_multi_company.users.ids:
-                user.write({'groups_id': [(4, group_multi_company.id)]})
-        return user
+    @api.model_create_multi
+    def create(self, vals_list):
+        new_vals_list = []
+        for values in vals_list:
+            new_vals_list.append(self._remove_reified_groups(values))
+        users = super(UsersView, self).create(new_vals_list)
+        group_multi_company_id = self.env['ir.model.data'].xmlid_to_res_id(
+            'base.group_multi_company', raise_if_not_found=False)
+        if group_multi_company_id:
```

Comment:
The `and 'company_ids' in values` is no longer there. Before it was not triggered if not setting explicitly any company, now it will as the default is `default=lambda self: self.env.company.ids`. In the interface it's not a problem as `company_ids` is present in the values but in case of mass-import, it will differ. Not saying the the old behaviour is better, just to double check.

Reply:
@tivisse Can we have your pov here ?

IMO, we should do the check even after considering the global defaults (which wasn't the case before).  If they say that by default all users have access to two companies, those users won't have the multi-company group if not created by the web-client.